### PR TITLE
Cloudsearch implementation

### PR DESCRIPTION
**Pull request by KingBrewer**
- implements Domain Configuration API of CloudSearch service
- implements Documents Batch Upload API of CloudSearch service

### CODE REVIEW

**Comment by waisbrot** [comment]
The Travis failure seems like it was some ephemeral problem with Hex.

File: src/erlcloud_aws.erl [hunk 1]
```diff
@@ -259,6 +259,7 @@ param_list(Values, Key) ->
     [{lists:flatten([Key, $., integer_to_list(I)]), Value} ||
         {I, Value} <- lists:zip(lists:seq(1, length(Values)), Values)].
 
+value_to_string(Float) when is_float(Float) -> io_lib:format("~p", [Float]);
```
Comment (waisbrot) [inline src/erlcloud_aws.erl]: For me, this yields a list of a string. Why not `float_to_list(Float)`?

Comment (motobob) [inline src/erlcloud_aws.erl]: and what does AWS API say about float format?

**Comment by motobob** [comment]
overall 👍 . sorry for not merging it promptly in AL fork.

File: src/erlcloud_aws.erl (hunk 1 above)
Comment (KingBrewer) [inline src/erlcloud_aws.erl]: @waisbrot Initially I was skeptical to `float_to_list`
```erlang
1> float_to_list(4.1).
"4.09999999999999964473e+00"
```
but AWS API handles it pretty well. I modified it as you suggested.

**Comment by KingBrewer** [comment]
@motobob comments addressed. The winner is `float_to_list/1`.

**Pull request merged**